driver/src/db_interface.rs

```diff
@@ -131,6 +135,38 @@ impl DbInterface for MongoDB {
             .map(|d| d.map(models::StandingOrder::from).map_err(DriverError::from))
             .collect::<Result<Vec<_>, _>>()
     }
+    fn get_standing_orders_index_of_slot(
+        &self,
+        slot: u32,
+    ) -> Result<Vec<U128>, DriverError> {
+        let pipeline = vec![
+            doc!{"$match" => (doc!{"validFromAuctionId" => (doc!{ "$lte" => slot})})},
+            doc!{"$sort" => (doc!{"validFromAuctionId" => -1, "_id" => -1, "accountId" => -1})},
+            doc!{"$group" => (doc!{"_id" => "$accountId", "batchIndex" => (doc!{"$first" => "$batchIndex" })})}
+        ];
+        
+        let standing_order_index_raw = self.client
+            .db(models::DB_NAME)
+            .collection("standing_orders")
+            .aggregate(pipeline, None)?
+            .map(|doc| doc.unwrap())
+            .map(|doc| (doc.get_i32("_id").unwrap() as u16,U128::from_dec_str(&(doc.get_i32("batchIndex").unwrap() as u16).to_string()).unwrap()))
+            //batchIndex is stored as i32. Probably, we should then also go with u32 in contract instead of U128
+            .collect::<Vec<_>>();
+
+        let mut standing_order_index = Vec::<U128>::with_capacity(50);
+        for i in 0..models::NUM_RESERVED_ACCOUNTS {
```

Comment:
What is the best practice? Can I have "so much" code in the db_interface? Should I pull it into order_driver? On the other hand its just inserting 0 entries...

Reply:
I'd suggest we expose the batchIndex on the `StandingOrder` struct that we are already parsing in the other method. The list of StandingOrders is already available in the driver and if we can access `batchIndex` on each of them we don't need this new method at all.

Reply:
You wrote the sql query in such a way that `batchIndex` is not accessible. It only gives you back:
`Standing Orders: [StandingOrder { account_id: 3, orders: [Order { account_id: 3, sell_token: 2, buy_token: 2, sell_amount: 5000000000000000000, buy_amount: 4000000000000000000 }] }]`

I could change your query. But I think these kinds of optimizations of the query for general usage might be wasted love, as new features will require even different query optimizations.

How do you feel about it? I really don't have a strong opinion here.

Reply:
Hm... I decided that I will refactor your query, now.

Reply:
> wasted love

What could this be!?

Reply:
Sorry just saw this. I believe all you'd need is another element in the group clause like:

```rust
"batchIndex" => (doc!{"$first" =>"$batchIndex" })
```